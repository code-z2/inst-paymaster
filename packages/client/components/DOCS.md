# Paymaster

This is a description of the components , stylesheets and
pages that makes up paymaster project.

## Components

This project makes use of tailwindcss and css modules.

Each component is styled using a local stylesheet. This makes it easy to identify where to make changes for any section or component in the page.

The navbar/footer component and other global variables are styled using a global stylesheet.

Components bear similar names to it's stylesheet modules
for easy identification.

E.g `landingpage/firstSection` styling can be seen in `styles/landing/firstsection.module.css`

Adopted Css grid method [Multi-grid-one-page-layout](https://medium.com/@nikkipantony/multi-grid-one-page-layout-css-grid-6efefd537404);

The components in this directory are:

## Active paymaster

- **activeNav**: This is the navbar of the active paymaster's page.
  `activepaymaster/activeNav`
  - styling: open `styles/navbar.css` to view the navbar styling.
    **All the navbars in the project are styled from**
    `styles/navbar.css`

```
 .active-paymaster-nav {
    @apply paynav;
    }

.active-paymaster-nav .list-ul {
@apply col-start-[-1] w-full grid grid-cols-[1fr_2fr] px-1 content-center text-xs sm:text-sm gap-10 font-light sm:font-normal;
}

```

    The`active-paymaster-nav ` class is reusing the same styling with the navbar in the landing page .

To view the paymasters page navigate to `http://localhost:3000/auth/{auth_id}/activepaymaster`

- **allpaymasters** : This is the paymasters list component.
  `activepaymaster/allpaymasters` styling for this component and the media queries is found at `styles/activepaymaster/allpaymasters.module.css`

- **firstSection** : It is the first part i.e first section of the active paymasters page

- **paymastersListPage** : This is where all the components in the active paymaster directory are assembled together and then exported to the active paymasters route.
  `activepaymaster/paymastersListPage` is exported to `/auth/[private]/activepaymaster`

## Auth directory

- **private**: A HOC for authentication using NEXT_PUBLIC_AUTH_ID environmental variable

## landing page

- **firstSection** : This is the hero section of the landing page.

- **landing** : This is where all the components in the landing page directory are assembled together and then exported to the landing page route.
  `activepaymaster/landingpage` is exported to `/auth/[private]`

**Applied padding for windows scaling b/w 100 to 125**

```

/*************scaling 100 to 125***********************/

@media (min-width: 1100px) and (-webkit-min-device-pixel-ratio: 1.25) {
  .scaleto100 {
    @apply px-20;
  }
}


```

`tailwind.config.css` custom configurations

```

 theme: {
    extend: {
      colors: {
        backgroundTheme: "#11142b",
        secondaryGlow: "#7d0cc1",
        jsClientGlow: "#7D0CC1",
        jsClientDark: "rgba(125, 12, 193, 0.3)",
        blurColor: "rgba(125, 12, 193, 0.5)",
        flowerBlur: "rgba(193, 12, 12, 0.5)",
      },
      fontFamily: {
        monserrat: "Montserrat",
        color: "#fff",
      },
      shadow: {
        btn: "4px 4px 14px rgba(0, 0, 0, 0.15)",
      },


```
