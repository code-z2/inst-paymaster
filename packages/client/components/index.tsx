// Reusable exports
export { default as Image } from "next/image";
export { default as Link } from "next/link";

// Landing page components
export { default as LandingPage } from "./landingpage/Default";

export { default as LandingPageLayout } from "./layouts/Default";
export { UnSecureLandingPage } from "./landingpage/Default";
export { default as Hero } from "./landingpage/Hero";
export { default as RebatesMiddleware } from "./landingpage/RebatesMiddleware";
export { default as BuildPaymasters } from "./landingpage/BuildPaymasters";

// authentication HOC component
export { default as Auth } from "./auth/private";

// paymasters list page components
export { default as PaymastersList } from "./activepaymaster/paymastersListPage";
export { default as Loadmore } from "./activepaymaster/loadmore";
export { default as ActiveNav } from "./activepaymaster/activeNav";
export { default as FirstSection } from "./activepaymaster/firstSection";
export { default as Allpaymasters } from "./activepaymaster/allpaymasters";
