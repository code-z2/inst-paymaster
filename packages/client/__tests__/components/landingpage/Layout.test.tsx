import React from "react";
import { useRouter } from "next/router";
import { render, screen } from "@testing-library/react";
import mockRouter from "next-router-mock";
import { LandingPageLayout, Hero } from "../../../components";

jest.mock("next/router", () => require("next-router-mock"));
/*
 *  snapshots of the Layout of the landing page components
 */
describe("Layout of the Landing page ", () => {
  it("mocks the useRouter hook", () => {
    // Set the initial url:
    mockRouter.push("/");
  });
  it("should render the child components ", () => {
    const home = render(
      <LandingPageLayout>
        {" "}
        <Hero />{" "}
      </LandingPageLayout>
    );
    expect(home).toMatchSnapshot();
  });
});
