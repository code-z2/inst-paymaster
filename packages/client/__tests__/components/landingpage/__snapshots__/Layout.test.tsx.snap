// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout of the Landing page  should render the child components  1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="site-header"
      >
        <div
          class="wrapper"
        >
          <div
            class="site-logo"
          >
            <img
              alt="Paymaster Logo"
              data-nimg="1"
              decoding="async"
              height="28"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
              style="color: transparent;"
              width="28"
            />
            <span
              class="logo-text"
            >
              Paymasters
            </span>
          </div>
          <nav
            class="site-nav "
          >
            <div
              class="wrapper"
            >
              <ul
                class="nav-list"
              >
                <li
                  class="nav-item"
                >
                  <a
                    href="/dev"
                  >
                    Docs
                  </a>
                </li>
                <div
                  class="nav-item"
                >
                  <a
                    href="/"
                  >
                    <button
                      class="cta"
                    >
                      Launch App
                    </button>
                  </a>
                </div>
              </ul>
            </div>
          </nav>
          <button
            class="cta site-nav-btn"
          >
            <svg
              aria-hidden="true"
              class="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 9h16.5m-16.5 6.75h16.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </button>
        </div>
      </header>
      <main>
         
        <section
          class="hero-section site-section"
        >
          <div
            class="wrapper w-image"
          >
            <div
              class="hero-content section-content"
            >
              <h1
                class="font-semibold text-5xl leading-normal"
              >
                The only paymasters access infrastructure you will need
              </h1>
              <p>
                Using paymasters to simplify zk transactions and provide superior UX to dapps.
              </p>
              <a
                href="/"
              >
                <button
                  class="cta"
                >
                  Launch App
                </button>
              </a>
            </div>
            <div
              class="hero-img-cont section-img-cont"
            >
              <img
                alt="Paymaster Hero Illustration"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </div>
          </div>
        </section>
         
      </main>
      <footer
        class="site-footer site-section"
      >
        <div
          class="wrapper"
        >
          <section
            class="logo-form-section"
          >
            <div
              class="site-logo"
            >
              <img
                alt="Paymaster Logo"
                data-nimg="1"
                decoding="async"
                height="28"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
                style="color: transparent;"
                width="28"
              />
              <span
                class="logo-text"
              >
                Paymasters
              </span>
            </div>
            <form
              class="newsletter-form"
            >
              <div
                class="form-header"
              >
                Subscribe to our news letter
              </div>
              <div
                class="form-group"
              >
                <div
                  class="form-control"
                >
                  <input
                    class="form-input"
                    placeholder="Enter your Email"
                    type="text"
                  />
                </div>
                <button
                  class="cta"
                >
                  Subscribe
                </button>
              </div>
            </form>
          </section>
          <section
            class="links-section"
          >
            <p
              class="copy"
            >
              â’¸ 2023 Paymasters. All rights reserved
            </p>
            <div
              class="links"
            >
              <ul
                class="socials links-list"
              >
                <li
                  class="link-item"
                >
                  <a
                    href="#"
                  >
                    Discord
                  </a>
                </li>
                <li
                  class="link-item"
                >
                  <a
                    href="#"
                  >
                    Telegram
                  </a>
                </li>
                <li
                  class="link-item"
                >
                  <a
                    href="#"
                  >
                    Twitter
                  </a>
                </li>
                <li
                  class="link-item"
                >
                  <a
                    href="#"
                  >
                    GitHub
                  </a>
                </li>
              </ul>
              <ul
                class="site links-list"
              >
                <li
                  class="link-item"
                >
                  <a
                    href="/dev"
                  >
                    Contact
                  </a>
                </li>
                <li
                  class="link-item"
                >
                  <a
                    href="/dev"
                  >
                    Privacy
                  </a>
                </li>
              </ul>
            </div>
          </section>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <header
      class="site-header"
    >
      <div
        class="wrapper"
      >
        <div
          class="site-logo"
        >
          <img
            alt="Paymaster Logo"
            data-nimg="1"
            decoding="async"
            height="28"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
            style="color: transparent;"
            width="28"
          />
          <span
            class="logo-text"
          >
            Paymasters
          </span>
        </div>
        <nav
          class="site-nav "
        >
          <div
            class="wrapper"
          >
            <ul
              class="nav-list"
            >
              <li
                class="nav-item"
              >
                <a
                  href="/dev"
                >
                  Docs
                </a>
              </li>
              <div
                class="nav-item"
              >
                <a
                  href="/"
                >
                  <button
                    class="cta"
                  >
                    Launch App
                  </button>
                </a>
              </div>
            </ul>
          </div>
        </nav>
        <button
          class="cta site-nav-btn"
        >
          <svg
            aria-hidden="true"
            class="icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.75 9h16.5m-16.5 6.75h16.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
      </div>
    </header>
    <main>
       
      <section
        class="hero-section site-section"
      >
        <div
          class="wrapper w-image"
        >
          <div
            class="hero-content section-content"
          >
            <h1
              class="font-semibold text-5xl leading-normal"
            >
              The only paymasters access infrastructure you will need
            </h1>
            <p>
              Using paymasters to simplify zk transactions and provide superior UX to dapps.
            </p>
            <a
              href="/"
            >
              <button
                class="cta"
              >
                Launch App
              </button>
            </a>
          </div>
          <div
            class="hero-img-cont section-img-cont"
          >
            <img
              alt="Paymaster Hero Illustration"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </div>
        </div>
      </section>
       
    </main>
    <footer
      class="site-footer site-section"
    >
      <div
        class="wrapper"
      >
        <section
          class="logo-form-section"
        >
          <div
            class="site-logo"
          >
            <img
              alt="Paymaster Logo"
              data-nimg="1"
              decoding="async"
              height="28"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
              style="color: transparent;"
              width="28"
            />
            <span
              class="logo-text"
            >
              Paymasters
            </span>
          </div>
          <form
            class="newsletter-form"
          >
            <div
              class="form-header"
            >
              Subscribe to our news letter
            </div>
            <div
              class="form-group"
            >
              <div
                class="form-control"
              >
                <input
                  class="form-input"
                  placeholder="Enter your Email"
                  type="text"
                />
              </div>
              <button
                class="cta"
              >
                Subscribe
              </button>
            </div>
          </form>
        </section>
        <section
          class="links-section"
        >
          <p
            class="copy"
          >
            â’¸ 2023 Paymasters. All rights reserved
          </p>
          <div
            class="links"
          >
            <ul
              class="socials links-list"
            >
              <li
                class="link-item"
              >
                <a
                  href="#"
                >
                  Discord
                </a>
              </li>
              <li
                class="link-item"
              >
                <a
                  href="#"
                >
                  Telegram
                </a>
              </li>
              <li
                class="link-item"
              >
                <a
                  href="#"
                >
                  Twitter
                </a>
              </li>
              <li
                class="link-item"
              >
                <a
                  href="#"
                >
                  GitHub
                </a>
              </li>
            </ul>
            <ul
              class="site links-list"
            >
              <li
                class="link-item"
              >
                <a
                  href="/dev"
                >
                  Contact
                </a>
              </li>
              <li
                class="link-item"
              >
                <a
                  href="/dev"
                >
                  Privacy
                </a>
              </li>
            </ul>
          </div>
        </section>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
